_vector-search/index.md

```diff
@@ -51,13 +51,1093 @@ items:
+
+{% include youtube-player.html id='oX0HMAztP8E' %}
+
+To follow the demo, use these steps.
+
+<details markdown="block">
+  <summary>
+    Steps
+  </summary>
+  {: .fs-5 .fw-700}
+
+### Prerequisites
+{:.no_toc} 
+
+Download the sample data for this demo:
+
+```bash
+wget https://amazon-pqa.s3.amazonaws.com/amazon_pqa_headsets.json
+```
+{% include copy.html %}
+
+Prepare data for bulk indexing into OpenSearch:
+
+```bash
+head -n 5000 amazon_pqa_headsets.json |  awk '{ print "{\"index\":{\"_index\":\"neural_search_pqa\"}}"; print;}'  > neural_search_amazon_pqa_headsets.json
+```
+{% include copy.html %}
+
+Enable running ML models on data nodes (not recommended for production environments):
+
+```json
+PUT /_cluster/settings
+{
+  "persistent": {
+    "plugins.ml_commons.only_run_on_ml_node": false
+  }
+}
+```
+{% include copy-curl.html %}
+
+### Step 1: Register and deploy a model
+{:.no_toc} 
+
+Register and deploy a machine learning (ML) model provided by OpenSearch:
+
+```json
+POST /_plugins/_ml/models/_register?deploy=true
+{
+	"name": "huggingface/sentence-transformers/all-distilroberta-v1",
+	"version": "1.0.1",
+	"model_format": "TORCH_SCRIPT"
+}
+```
+{% include copy-curl.html %}
+
+Registering a model is an asynchronous task. OpenSearch returns a task ID for this task. Check the status of the task by using the Tasks API:
+
+```json
+GET /_plugins/_ml/tasks/<task_id>
+```
+{% include copy-curl.html %}
+
+Once the task is complete, the task state will change to `COMPLETED` and the Tasks API response will contain a model ID for the registered model. Note the model ID; you'll use it in the following steps.
+
+### Step 2: Create an ingest pipeline 
+{:.no_toc} 
+
+Create an ingest pipeline that will generate vector embeddings from text:
+
+```json
+PUT _ingest/pipeline/nlp-index-pipeline
+{
+  "processors" : [
+    {
+      "text_embedding": {
+        "model_id": "<model_id>",
+        "field_map": {
+          "question_text": "question_vector"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Test the ingest pipeline:
+
+```json
+POST /_plugins/_ml/_predict/text_embedding/<model_id>
+{
+  "text_docs":[ "what does the package contain?"],
+  "return_number": true,
+  "target_response": ["sentence_embedding"]
+}
+```
+{% include copy-curl.html %}
+
+### Step 3: Create an index
+{:.no_toc} 
+
+Create a vector index and set the default ingest pipeline to the ingest pipeline created in the previous step:
+
+```json
+PUT /neural_search_pqa
+{
+  "settings": {
+    "index.knn": true,
+    "default_pipeline": "nlp-index-pipeline"
+  },
+  "mappings": {
+    "properties": {
+      "question_vector": {
+        "type": "knn_vector",
+        "dimension": 768
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+### Step 4: Ingest data
+{:.no_toc} 
+
+Ingest the data you prepared in Prerequisites:
```

Comment:
Either "Ingest the data you prepared in the [Prerequisites](link) section" or "Ingest the data you prepared as part of the [Prerequisites](link) for this tutorial" (include link to section).